src/plugins/intel_npu/src/backend/src/zero_infer_request.cpp

```diff
@@ -388,6 +388,10 @@ void ZeroInferRequest::set_tensor(const ov::Output<const ov::Node>& port, const
     }
 
     if (foundPort.is_input()) {
+        if (get_user_input(foundPort.idx)._ptr == tensor._ptr) {
```

Comment:
I think we may need to check first if user input exist

Reply:
Forgot that right now we create them already when infer is created and set them to nullPtr until are allocated. So this might work. Could you just check for nullptr as well, please?

If user tensor != nullptr and user tensor == tensor

Reply:
if `user_tensor` is nullptr, then, depending on `tensor`
- if the latter is not nullptr, then `user_tensor` == `tensor` will be `false` so no early return done - continue and update
- if the latter is `nullptr`, they are both `nullptr`s so nothing to be done?

Reply:
Yeah, that is also true.